src/ratings-and-reviews/components/AddReview.jsx

```diff
@@ -0,0 +1,292 @@
+import React from 'react';
+import { Modal, validateEmail} from '../../common';
+import { Formik, Form, Field, FieldArray, ErrorMessage } from 'formik';
+import '../ratings-and-reviews.scss';
+import api from '../../api';
+
+const ratingConverter = (rating) => {
+  if (rating === 'Great') {
+    return 5;
+  }
+  if (rating === 'Good') {
+    return 4;
+  }
+  if (rating === 'Average') {
+    return 3;
+  }
+  if (rating === 'Fair') {
+    return 2;
+  }
+  if (rating === 'Poor') {
+    return 1;
+  }
+  if (rating === 'Yes') {
+    return true;
+  }
+  if (rating === 'No') {
+    return false;
+  }
+};
+
+const AddReview = ({hide, product, characteristics}) => {
+
+  let [textCount, setTextCount] = React.useState(0);
+
+  const characteristicNames = Object.keys(characteristics);
+  const initialValues = {rating: '', recommend: '', summary: '', body: '', nickname: '', photos: [], email: ''};
+  for (const characteristic of characteristicNames) {
+    initialValues[characteristic] = '';
+  }
+  return (
+    <Modal hide={hide}>
+      <h1>Write Your Review</h1>
+      <h2>About the {product.name}</h2>
+      <Formik
+        initialValues={initialValues}
+        validate={values => {
+          const errors = {};
+          if (!values.rating) {
+            errors.rating = 'Required';
+          }
+          if (!values.recommend) {
+            errors.recommend = 'Required';
+          }
+          if (!values.body) {
+            errors.body = 'Required';
+          }
+          if (!values.nickname) {
+            errors.body = 'Required';
+          }
+          if (!values.email) {
+            errors.email = 'Required';
+          } else if (!validateEmail(values.email)) {
+            errors.email = 'Invalid email address';
+          }
+          return errors;
+        }}
+        onSubmit={(values, {resetForm, setSubmitting}) => {
+          const postConstructor = {
+            rating: ratingConverter(values.rating),
+            summary: values.summary || '',
```

Comment:
I might be missing something, but I don't think you need that `|| ''`. You already set `summary` to start as `''` from `initialValues`.